packages/graph/hash_graph/lib/graph/src/api/rest/utoipa_typedef/subgraph/edges.rs

```diff
@@ -56,7 +77,42 @@ impl From<crate::subgraph::edges::Edges> for Edges {
             knowledge_graph: KnowledgeGraphRootedEdges(edges.knowledge_graph.into_iter().fold(
                 HashMap::new(),
                 |mut map, (id, edges)| {
-                    let edges = edges.into_iter().collect();
+                    let edges = edges.into_iter().map(|edge| {
+                        match edge {
+                            crate::subgraph::edges::KnowledgeGraphOutwardEdges::ToOntology(edge) => {
+                                KnowledgeGraphOutwardEdges::ToOntology(edge)
+                            }
+                            crate::subgraph::edges::KnowledgeGraphOutwardEdges::ToKnowledgeGraph(
+                                edge,
+                            ) => {
+                                let earliest_timestamp = if edge.reversed {
+                                    // We want to log the time the link entity was *first* added
+                                    // from this entity. We therefore need to find the timestamp of
+                                    // the first link entity
+                                    vertices.find_earliest_entity(&edge.right_endpoint)
```

Comment:
If we stored the vertices in a btree-map we wouldn't need to do a search. In bad cases this is also pretty inefficient, it's two loops (although I appreciate the inner loop is only over the editions of that entity)

Reply:
That's a good point, I changed the representation to use a `BTreeMap` instead: 14938b548c5dea70a4e51bcd8f3a94cc3e00b5c8